seed/views/v3/analyses.py

```diff
@@ -0,0 +1,71 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2020, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+from drf_yasg.utils import swagger_auto_schema
+from django.http import JsonResponse
+from rest_framework import viewsets
+from rest_framework.status import HTTP_409_CONFLICT
+
+from seed.decorators import ajax_request_class, require_organization_id_class
+from seed.lib.superperms.orgs.decorators import has_perm_class
+from seed.models import Analysis
+from seed.serializers.analyses import AnalysisSerializer
+from seed.utils.api import api_endpoint_class
+from seed.utils.api_schema import AutoSchemaHelper
+
+
+class AnalysisViewSet(viewsets.ViewSet):
+    serializer_class = AnalysisSerializer
+    model = Analysis
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            AutoSchemaHelper.query_org_id_field(True),
+            AutoSchemaHelper.query_integer_field('property_id', True, 'Property ID')
+        ]
+    )
+    @require_organization_id_class
+    @api_endpoint_class
+    @ajax_request_class
+    @has_perm_class('requires_member')
+    def list(self, request):
+        organization_id = request.query_params.get('organization_id', None)
+        property_id = request.query_params.get('property_id', None)
+        analyses = []
+        for analysis in Analysis.objects.filter(organization=organization_id):
+            property_view_info = analysis.getPropertyViewInfo(property_id)
+            if property_id is not None and property_view_info["number_of_analysis_property_views"] < 1:
+                continue
+            serialized_analysis = AnalysisSerializer(analysis).data
+            serialized_analysis.update(property_view_info)
+            analyses.append(serialized_analysis)
```

Comment:
I think there might be a slightly simpler way to do this. Django has some powerful filtering abilities built in, see: https://docs.djangoproject.com/en/2.2/topics/db/queries/#spanning-multi-valued-relationships

What if we did something like this (not sure if this whole thing works but you get the idea)
```python
if property_id is not None:
    analyses_qs = Analysis.objects.filter(organization_id=organization_id, analysispropertyview__property_id=property_id)
else:
    analyses_qs = Analysis.objects.filter(organization_id=organization_id)
for analysis in analyses_qs:
    serialized_analysis = AnalysisSerializer(analysis).data
    serialized_analysis.update({
        'number_of_analysis_property_views': analysis.analysispropertyview_set.count(),
        'cycles': list(analysis.analysispropertyview_set.values_list('cycle', flat=True).distinct())
    })
    analyses.append(serialized_analysis)
```

Reply:
That would work for sure but the ticket states the number of views and cycles should be analysis model methods.  I built "getPropertyViewInfo" with this in mind so the analysisPropertyViews table wouldn't be hit more than once, even with Django's lazy loading.  I can split the difference and generate the property filtered queryset so there's no continue.  Does that work?

Reply:
I like the split the difference idea 👍 👍  I guess that would also mean we don't need the `property_id` param for your method, which I like as well

Reply:
I just realized (thanks tests!) that param is going to be required no matter how you look at it.  In order to get an accurate count of property views that method will need to know if you are filtering on a property id or not.